### ISSUE TITLE
Rome was built in 1776: A Case Study on Factual Correctness in Knowledge-Grounded Response Generation

### ISSUE THREAD

**Issue by dreamingjudith** [dreamingjudith/paper-reading-list#19]
[Arxiv](https://arxiv.org/abs/2110.05456)

# 0. 요약

- Knowledge-grounded response는 3가지 타입으로 나눌 수 있으며 factual consistency detector 모델을 소개한다
- Wizard of Wikipedia 데이터셋을 기반으로 한 Conv-FEVER라는 새로운 코퍼스를 구축했다
- 데이터셋은 공개할 예정이라는데 아직 안 됨 (2022년 1월 19일 현재)

# 1. 들어가기에 앞서

### Factual consistent

- Accurately portratys the **input knowledge** (assuming the provided knowledge input is correct)
- 본 논문에서 다루려는 부분

### Factual correct

- Accurate with respect to the **world knowledge**
- 이것도 중요하기는 한데 future work로 남기려고 한다

# 3. Are Neural Responses Consistent With Grounded Knowledge?

### Non-verifiable responses

- response에 information이 있다고 볼 수가 없기 때문에 애초에 평가 자체가 불가능

### Factually consistent responses

- response에 detailed information이 담겨 있음
- do not contradict the **input knowledge**

### Hallucinated responses

- information은 담겨 있음
- 그러나 input knowledge와 일맥상통하진 않음
    - 따라서 correct한지 아닌지도 모름
    - 그러나 이런 hallucination이 간간히 **world knowledge** 에 비춰봤을 때는 맞을 때도 있긴 함

![image](https://user-images.githubusercontent.com/20284375/150137536-2f2a6f1c-d6dd-4d5d-b200-680244895767.png)

## 3.1 Annotation Scheme

- We propose two-stage human annotation setup for factual correctness
- The annotators are given a dialog context along iwht a knowledge sentence, and evaluate the system generated response

### Stage 1

- Appropriateness: relevance to the dialog context
- Verifiability: to what degree the response needs to be verified
- 리커트 척도 1-5 사이로 점수를 매김

### Stage 2

- Factual consistency
    - Is the response generated factually accurate with regards to the input knowledge?
        - Facually incorrect → 0
        - Partially correct → 0.5
        - Completely correct → 1
- Hallucination
    - Is the response generated making up more information than what is provided in the conversational context and input knowledge?
        - Yes
        - No

### *Factual dataset-crowd*

- 각각 다른 neural response generator의 output을 crowd-workers를 데리고 어노테이션함 → 그래서 만든 데이터셋 이름이 *Factual dataset-crowd*
- 모델 크기, knowledge retrieval 방식이나 decoding step 등의 배리에이션을 줘서 setup에 따라 factual correctness가 얼마나 차이가 나는지 보려고 함

### 저자들이 실험을 위해 사용한 것들

- Wizard of Wikipedia dataset
    - 아마존 메카니컬 터크를 통해 수집된 데이터셋으로 2명이 각각 Wizard와 Apprentice 역할을 맡아 멀티턴 대화를 진행
    - Apprentice는 배우려는 입장이고, Wizard는 위키피디아에 접근할 수 있다는 컨셉.
    - Wizard는 Apprentice의 말에 따라 적절한 Knowledge-grounded response를 제공하기 위해 매 턴마다 위키피디아 문서에서 knowledge sentence를 선택
- GPT-2 model
    - TransferTransfo 방식으로 WoW dataset에서 파인튜닝함
    : Take in as input both the **dialog context** and a **knowledge sentence** and minimize cross-entropy loss on the **ground-truth response.**
    - 보캡은 BPE 기반

### 실험 조건

- Model size: small, medium, large, XL
- Decoding strategies
    - Nucleus sampling (NS) with p=0.9
    - Beam search (BS) with beam_size=5
    - Delayed beam search (DBS)
        - top-k sampling for n delay steps followed by beam search
        - k=10, n=5, beam_size=5
- Knowledge sentence configuration type
    - Ground Truth Knowledge (GT)
        - WoW 데이터셋에서 “Wizard”의 응답을 모아논 거기 때문에 WoW test set에서만 사용 가능
    - K-Nearest Neighbours (K-NN)
        - we use a K nearest neighbor search across all the 5 million Wikipedia articles used in the WoW dataset
        - knowledge base와 input dialog는 DistillBERT를 NLI와 STSb 태스크로 파인튜닝한 모델을 통해 인코딩
            - 우선 knowledge base를 만들기 위해 위키피디아 문서 내의 개별 문장마다 인코딩
            - input dialog에 대해서는 each turn마다 encoder 모델로 인코딩
        - take the average of the encoded embeddings and perform kNN search using the FAISS library
        - We retrieve the top most relevant sentence from the knowledge base
    - Dense Passage Retriever (DPR)
        - DPR 모델에서 knowledge base는 21 million 개의 위키피디아 문서 사용
        - Pretrained encoder model로 마지막 턴의 문장을 인코딩한 후, knowledge base로부터 the most relevant passage를 FAISS-based similarity search를 통해 찾음

### 데이터셋 구성

- WoW 데이터셋으로부터 5-turn짜리 대화 100개를 샘플링
- 위에서 설명한 실험 조건으로 모든 대화의 맨 마지막 response를 생성해냄
→ 100 dialogs * 4 model sizes * 3 knowledge setups * 3 decoding mechanisms = 3600 responses
- 3600개 대화에 대해 사람이 어노테이션함
    - Stage 1에서 Appropriateness 3 미만 먼저 제외한 다음 Verifiability 3 미만인 응답 제외
        - 3600 (original) → 2781 (appropriateness filtering) → 1869 (verifiability filtering)
        - appropriate response의 33%는 non-verifiable하다는 걸 보여줌
    - 이후 세팅별 평균 52개의 응답이 Stage 2에서 어노테이션됨
        - Hallucination: majority voting 적용
        - Inter-annotator agreement(IAA): Krippendorff’s alpha 사용

### 평가 결과

![image](https://user-images.githubusercontent.com/20284375/150137653-90e13075-d422-4b6e-bea2-8c919f1395af.png)

- Factual consistency
    - 어떤 SOTA 모델도 성능이 좋게 나오질 않음
    - 그러나 모델이 커질수록 facutal consistency가 늘고 hallucinated response가 줄긴 했다
    - 가장 많은 양의 knowledge base가 사용됐던 DPR 세팅이 factual consistency가 제일 높고 hallucination 비율은 제일 낮았다
    - DPR 환경에서 모델이 커질 수록 DBS가 잘 동작했다 (이전 연구에서는 Document-level generation에서만 성능을 확인함)
- Appropriateness score
    - NN(Nearest neighbours)가 DPR보다 성능이 잘 나옴
    - Input knowledge quality can affect both the response quality and the factual consistency in model responses

# 4. Factual Consistency Detector

### 개요

- FEVER dataset의 레이블링 방안에 착안해 WoW 데이터셋을 기반으로 구축
- WoW dataset에서 Wizard의 마지막 response가 factually consistent하다고 가정, data augmentation을 적용해 factually inconsistent response를 생성함

### Data augmentation 방법

- 범례
    - Conversational Context $C_n$
    - Knowledge $K_n$
    - Wizard’s Response or Target Response $R_n$
- Random Pairing
    - 서로 다른 Conversation에서 Response끼리 바꾸기
    - 서로 다른 Conversation에서 Knowledge끼리 바꾸기
- Negation
    - Response나 Knowledge 내에 특정 동사가 보이면 거기에 not/n’t 붙여서 반대로 만들어버리기
- Entity Swapping
    - spaCy NER tagger를 이용해 각각의 NE 위치 리스트 추출
        - Entities mentioned in conversational context ($E_c$)
        - Entities mentioned in knowledge ($E_k$)
    - Conversational context에서 Common entity 발견 시 $E_c$ 리스트에서 하나 랜덤 샘플링한 뒤 위치 맞바꾸기
    - Knowledge에서 Common entity 발견 시 $E_k$ 리스트에서 하나 랜덤 샘플링한 뒤 위치 맞바꾸기

### 4.2.1. Baseline: ALBERT-base

- ALBERT-base 모델을 세 가지 데이터셋에 파인튜닝 시킴
    - MNLI: Entailment task with 3 labels - entailment, neutral, contradiction
    - FEVER, VitaminC
        - Claim이 주어진 reference piece of knowledge에 의해 확인이 되는지를 체크하는 태스크
        - 3 labels - supports, not enough info, refutes
    - 그래서 MNLI의 레이블을 FEVER, VitaminC에서 사용한 레이블과 매핑시킴
        - entailment → supports (factually consistent)
        - neutral → not enough info (factually inconsistent)
        - contradiction → refutes (factually inconsistent)
    - 그러나 hallucination이 knowledge를 contradict나 refute한다고 볼 수는 없어서 not enough inf의 example만 factually inconsistent로 간주함

### 4.2.2. Baseline: FactCC

- BERT-based factual consistency prediction model for neural summarization models

### 4.2.3. BERT-base-Conv-FEVER

- BERT-base 모델을 백본으로 FEVER dataset(KILT에서 소개된 태스크)으로 학습함으로써 initialize를 시킨 다음 Conv-FEVER dataset으로 파인튜닝함

### 모델 성능

![image](https://user-images.githubusercontent.com/20284375/150141695-60983e2d-7a8d-4ab3-a667-8873ed099e1e.png)

# 5. Factual Consistency Detector Evaluation

- 앞서 3.2에서 봤던 *Factual dataset-crowd*와는 달리 ground truth knowledge만 가지고 데이터셋을 구성함 → 이름은 *Factual dataset-expert*
- 앞선 데이터셋의 inter-annotator agreement 수치가 낮았던 점 때문에 이 데이터셋을 사람이 어노테이션 할 때는 3명의 내부 어노테이터만 데리고 매김

## 데이터셋 구성

- fine-tuned GPT-2 기반 1000개 output을 사람이 어노테이션함
    - ground-truth knowledge
    - nucleus sampling with p=0.9
- 최종 response 개수: 584 (from 1000)
    - Stage 1에서 41.6%는 non-verifiable response로 매겨짐
    - Stage 2 때 3명의 어노테이터의 factual consistency를 평균낸 뒤 0.5 이상인 건 consistent, 미만인 건 inconsistent로 매김
    - 그랬더니 consistent 502개, inconsistent 82개로 매겨짐
    - 최종 결과물은 Krippendorff’s alpha 기반의 inter-annotator agreement score가 꽤 높았음
    (factual consistency 0.8, hallucination 0.86)
- 데이터셋 평가 결과

![image](https://user-images.githubusercontent.com/20284375/150141748-7b8b3faf-e9d6-4b7e-bbbf-a9416e331945.png)
